independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/producer/generic/RecursiveGenericInvalidTest.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.arc.test.producer.generic;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.arc.test.ArcTestContainer;
+
+public class RecursiveGenericInvalidTest {
+    @RegisterExtension
+    public ArcTestContainer container = ArcTestContainer.builder()
+            .beanClasses(Producer.class, Target.class)
+            .shouldFail()
+            .build();
+
+    @Test
+    public void test() {
```

Comment:
It's a good practice to assert the `container.getFailure()` in the test method :-)

Reply:
Ah, I don't remember seeing that anywhere, but in hindsight, that's obvious! I'll add a fix.

Reply:
Done.